### PR TITLE
Update release process to guide team practice

### PR DESCRIPTION
**Pull request by EoinShaughnessy**
Addresses [#2175](https://github.com/alphagov/govuk-frontend/issues/2175) (Map release process and identify pain points).

We recently had a retro to identify pain points in our release process.

This pull request contains revised steps, based on team feedback, to the doc for our release process. For example, that no major changes should go into a release after an agreed cutoff-point.

Hopefully these revisions will make our future releases run more smoothly.

### CODE REVIEW

**Comment by EoinShaughnessy** [comment]
Do we need to change `master` to `main` in this content? Or can that wait till we've completed the branch-renaming work?

**Comment by vanitabarrett** [comment]
@EoinShaughnessy I think we can do that separately - it might take us a while to do the rename and we probably don't want to update the docs until it's done, so I wouldn't want us to block this PR 🙂

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:3]: Would this work:

> Coordinate the final cutoff date by letting the GOV.UK Design System team know we are close to releasing at stand up. Once the agreed cut off date has passed, do not add any further major changes to the release.

It might help to clarify that by cutoff we mean a date rather than a process or other task.

Comment (hannalaakso) [inline docs/releasing/publishing.md:3]: I think the steps that might be repeated are now steps 4-6.

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:27]: I think this should say 'Content designer and designers'

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:31]: I think this should say 'Content designer/Community manager'

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:3]: It should be 'We can still add small fixes before step 7'

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:103]: Could we use the https://ukgovernmentdigital.slack.com/app_redirect?channel=govuk-design-system etc. syntax for linking to Slack channels? They're more readable and less likely to break in the future than the archive links.

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:26]: Broken link here.

File: docs/releasing/publishing.md
Comment (hannalaakso) [inline docs/releasing/publishing.md:45]: This is not to do with the changes in this PR but I just noticed a broken link here. This should be '/docs/contributing/versioning.md' 🔍

**Review by hannalaakso — COMMENTED** [review]
Thanks @EoinShaughnessy 🎉 I think these changes will really help us to streamline the release process. Noticed a couple of small things that just need tidying up.

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:3]:
```suggestion
1. At stand up, tell the GOV.UK Design System team we are close to releasing so we can coordinate the final cutoff date. Once the agreed cutoff date passes, do not add any further major changes to the release. We can still add small fixes before step 7 as long as we notify the content designer and technical writer. However, we should try to avoid adding too many fixes in this way, as it requires us to repeat some of steps 4-6.
```

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:26]:
```suggestion
7. Coordinate the sign-off when you are ready to do the release. Once the team agrees, this confirms a code and content freeze. Use the [#design-system-team-channel](https://gds.slack.com/app_redirect?channel=design-system-team-channel) to confirm sign-off from:
```

Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:26]: Have replaced old link with one that has more robust syntax.

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:27]: I think the intent here was that 'Content' is shorthand for 'content designer + technical writer.' Would it be ok to write "Content designer, technical writer and designers"?

Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:27]:
```suggestion
  - Content designer, technical writer and designers for guidance, examples and community backlog decision rationale
```

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:31]: As with the previous comment, 'Content' here was meant as shorthand for 'content designer + tech writer.' Would it be ok to write "Content designer/Community manager/Technical writer"? I don't always draft comms, but it's something I do occasionally (like with the ongoing branch-renaming work).

Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:31]:
```suggestion
  - Content designer/Community manager/Technical writer for announcements and engagement activities
```

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [reply docs/releasing/publishing.md:45]:
```suggestion
4. Create and checkout a new branch (`release-[version-number]`). See the [versioning documentation](/docs/contributing/versioning.md) for more information.
```

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [inline docs/releasing/publishing.md:103]:
```suggestion
  - GOV.UK Frontend in the govuk-design-system channels on [GDS Slack](https://gds.slack.com/app_redirect?channel=govuk-design-system) and [x-gov Slack](https://ukgovernmentdigital.slack.com/app_redirect?channel=govuk-design-system)
```

File: docs/releasing/publishing.md
Comment (EoinShaughnessy) [inline docs/releasing/publishing.md:104]:
```suggestion
  - Prototype kit in the [prototype-kit channel](https://ukgovernmentdigital.slack.com/app_redirect?channel=prototype-kit) on x-gov Slack
```

**Comment by EoinShaughnessy** [comment]
> Thanks @EoinShaughnessy 🎉 I think these changes will really help us to streamline the release process. Noticed a couple of small things that just need tidying up.

Thanks for the excellent review! I think I've actioned your feedback now. If things look ok, I'll add in the final commits and then get to squashing.

File: docs/releasing/publishing.md
Comment (hannalaakso) [reply docs/releasing/publishing.md:27]: Yes that makes sense 👍

File: docs/releasing/publishing.md
Comment (hannalaakso) [reply docs/releasing/publishing.md:31]: Yes, sounds good 👍

**Review by hannalaakso — APPROVED** [review]
Thanks for addressing all my comments @EoinShaughnessy, this looks good to me 🙏 I've given it an approval but squashing the commits before it's merged sounds great.

**Pull request merged**
